scipy/stats/_morestats.py

```diff
@@ -3373,6 +3377,178 @@ def func(x):
+    ...     68, 75, 68, 70, 72, 67, 70, 70, 69, 72, 71, 74, 75
+    ... ])
+
+    >>> from scipy.stats._morestats import f_equality_test
+    >>> args = [tenor1, bass2]
+    >>> res = f_equality_test(*args)
+    >>> res.statistic
+    1.4894234553325463
+    >>> res.pvalue
+    0.3430228457832443
+
+    If we assume alpha is 0.05, then we cannot reject the null hypothesis.
+    We have no good evidence from the data that the heights of Tenor1 and
+    Bass2 singers have different variances (despite there being a significant
+    difference in mean heights of about 2.5 inches.)
+
+    In [3]_, A forester wants to compare two different mist blowers for
+    consistent application. She wants to use the mist blower with the
+    smaller variance, which means more consistent application.
+    She wants to test that the variance of Type A (0.087 gal.2) is
+    significantly greater than the variance of Type B (0.073 gal.2)
+    using a level of significance of 0.05.
+
+    Type A has a sample of 16 mist blowers with a variance of 0.079.
+    Type B has a sample of 21 mist blowers with a variance of 0.060.
+
+    >>> import numpy as np
+    >>> typeA = np.array([
+    ...     11.2, 11, 11.3, 11.5, 11, 11, 11.5, 11.3, 10.9,
+    ...     11.2, 10.9, 10.9, 11.1, 10.5, 10.6, 10.9
+    ... ])
+    >>> typeB = np.array([
+    ...     10.8, 11.2, 10.9, 10.7, 11.5, 11, 11.1, 10.7, 11, 11.1, 10.8,
+    ...     11.2, 10.9, 11, 10.9, 11.6, 11.1, 10.8, 11.3, 10.8, 11.2
+    ... ])
+
+    >>> from scipy.stats._morestats import f_equality_test
+    >>> args = [typeA, typeB]
+    >>> res = f_equality_test(*args, alternative='greater')
+    >>> res.statistic
+    1.3079968329374523
+    >>> res.pvalue
+    0.7169386351928994
+
+    The test statistic is not larger than the critical value (it does not
+    fall in the rejection zone) so we fail to reject the null hypothesis.
+    While the variance of Type B is mathematically smaller than the variance
+    of Type A, it is not statistically smaller. There is not enough statistical
+    evidence to support the claim that the variance of Type A is significantly
+    greater than the variance of Type B. Both mist blowers will deliver the
+    chemical with equal consistency.
+
+    """
+    k = len(samples)
+    if k != 2:
+        raise ValueError("Must enter two input sample vectors.")
+    
+    if alternative not in {'two-sided', 'greater', 'less'}:
+        raise ValueError("'alternative' must be 'two-sided',"
+                         " 'greater', or 'less'.")
+    
+    # Handle empty input and input that is not 1d
+    for sample in samples:
+        if np.asanyarray(sample).size == 0:
+            NaN = _get_nan(*samples)  # get NaN of result_dtype of all samples
+            return FResult(NaN, NaN)
+
+    Ni = np.empty(k)
+    ssq = np.empty(k, 'd')
+    for j in range(k):
+        Ni[j] = len(samples[j])
+        ssq[j] = np.var(samples[j], ddof=1)
```

Comment:
Instead of looping, is it not possible to use the `axis` argument of [np.var](https://numpy.org/doc/stable/reference/generated/numpy.var.html)?